source/common/http/http1/codec_impl.cc

```diff
@@ -411,8 +411,9 @@ int ServerConnectionImpl::onHeadersComplete(HeaderMapImplPtr&& headers) {
         http_parser_pause(&parser_, 1);
       }
 
-    } else if (parser_.method == http_method::HTTP_POST ||
-               parser_.method == http_method::HTTP_PUT) {
+    } else if ((parser_.method == http_method::HTTP_POST ||
+                parser_.method == http_method::HTTP_PUT) &&
+               headers->ContentLength() == nullptr) {
```

Comment:
"IIRC there is some security considerations to requiring Content-Length as a mitigation for HTTP Request Smuggling attacks."
This.  But as we're not using it in production yet we can do it the slow and save way of adding stats and making sure it won't do harm.  Or alternately making sure we add an explicit content-length:0 so all layers agree.

Either way sounds like whatever we do should be done in a follow-up so I'll remove it here :-)